docs/adr/0004-runtime-governance.md

```diff
@@ -0,0 +1,225 @@
+  provide stake in escrow.
+
+  Runtimes using this governance model cannot be registered via regular registry
+  method calls (doing so must return `ErrForbidden`). Instead such a runtime can
+  only be registered at genesis, through a network upgrade or via a consensus
+  layer governance mechanism not specified by this proposal.
+
+### State
+
+This proposal introduces/updates the following consensus state in the registry
+module:
+
+#### Stored Runtime Descriptors
+
+Since the runtime descriptors can now be updated by actors other than the
+initial registering entity, it does not make sense to store signed runtime
+descriptors. The value of storage key prefixed with `0x13` which previously
+contained signed runtime descriptors is modified to store plain runtime
+descriptors.
+
+#### Pending Runtime Updates
+
+Key format:
+
+| Field         | Type                           | Content                    |
+|:--------------|:------------------------------:|:--------------------------:|
+| prefix        | `0x1A` (1 byte)                | state key prefix           |
+| H(runtime-id) | H(common.Namespace) (32 bytes) | runtime identifier         |
+
+Value:
+
+```golang
+type PendingRuntimeUpdate struct {
+    Epoch      epochtime.EpochTime `json:"epoch"`
+    Descriptor registry.Runtime    `json:"descriptor"`
+}
+```
+
+### Transaction Methods
+
+This proposal updates the following transaction methods in the registry module:
+
+#### Register Runtime
+
+Runtime registration enables a new runtime to be created or an existing runtime
+to be updated (in case the governance model allows it).
+
+**Method name:**
+
+```
+registry.RegisterRuntime
+```
+
+The body of a register runtime transaction must be a `Runtime` descriptor.
+The signer of the transaction must be the owning entity key.
+
+Registering a runtime may require sufficient stake in either the owning entity's
+(when entity governance is used) or the runtime's (when runtime governance is
+used) escrow account.
+
+### Messages
+
+This proposal introduces the following runtime messages:
+
+#### Update Runtime Descriptor
+
+The update runtime descriptor message enables a runtime to update its own
+descriptor when the current governance model allows it.
+
+**Field name:**
+
+```
+update_runtime
+```
+
+**Body:**
+
+```golang
+type UpdateRuntimeMessage struct {
+    registry.Runtime
+}
+```
+
+The body of the update runtime descriptor message is a new runtime descriptor
+that must be for the runtime emitting this message. Otherwise the message is
+considered malformed.
+
+### Consensus Parameters
+
+#### Registry
+
+This proposal introduces the following new consensus parameters in the registry
+module:
+
+- `disable_runtime_governance_models` (set of `RuntimeGovernanceModel`)
+  specifies the set of runtime governance models that are not allowed to be used
+  when updating registrations (either via method calls or via runtime messages).
```

Comment:
or when creating registrations?

Reply:
The create/update is the same method call, but yeah we can say "when calling `registry.RegisterRuntime`" to be more clear instead.